### DISCUSSION TITLE
The case for "Update Component" Action

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7853] [Ideas]
The current way of interacting with components and changing properties of them, is reactive.
We have to set a State variable, and then using a Conditional UI rule act accordingly.
This is a very useful feature.

Now, i suggest there should also be an active way, via an Action.

Scenario : 
Button -> OnClick
 "Update Component" Action 
  - > Set Icon to "loading"
  - > Set container height to 500px
  - > Set Button to disabled.
  - > Set Image Custom CSS to " ... "

To achieve the same result, using the reactive approach, would take setting a State variable and then 4 Conditional UI rules on 4 different Components. Which makes it complicated to maintain later and keep track of this logic.

With the active approach , it would take 1 Action, with 4 parameters easy to understand and maintain.

Thoughts ?

**Comment by melohagan** [comment]
I don't see any reason why not! Feature request here: https://github.com/Budibase/budibase/issues/7979
